tools/triton-eqm-create.sh

```diff
@@ -0,0 +1,495 @@
+        printf 'Prefix length %s out of range.\n' "${subnet[1]}"
+        return 1
+    fi
+
+    case "$which_end" in
+        start)
+            octet[3]=$(( octet[3] + 2 )) ;;
+        end)
+            octet[3]=$(( octet[3] + ${prefix_num_ips[$prefix_l]} - 2)) ;;
+        *)
+            printf 'can only get start or end ip\n'
+            return 1
+            ;;
+    esac
+
+    printf '%s.%s.%s.%s\n' "${octet[0]}" "${octet[1]}" "${octet[2]}" \
+        "${octet[3]}"
+}
+
+function show_napi_commands
+{
+    local s_hostname="$1"
+    local server="$2"
+    local ip_addresses network_name nic_tag obj
+    local provision_end provision_start
+    local network prefix_l
+
+    nic_tag="external_rack_${s_hostname}r"
+    network_name="external_rack_${s_hostname}r"
+
+    json_tmpl='{
+  "nic_tag": "%s",
+  "name": "%s",
+  "subnet": "%s/%s",
+  "gateway": "%s",
+  "provision_start_ip": "%s",
+  "provision_end_ip": "%s",
+  "vlan_id": 0,
+  "resolvers": [
+    "8.8.8.8",
+    "8.8.4.4"
+  ],
+  "description": "External for %s"}'
+
+    ip_addresses=$(packet server get -i "$server" -j | json ip_addresses | \
+        json -o jsony-0 -ac 'this.public==true')
+
+    printf 'sdc-napi /nic_tags -X POST -d '
+    printf \'
+    printf '{"name":"%s"}' "$nic_tag"
+    printf \''\n'
+
+    while read -r obj; do
+        gateway=$(json gateway <<< "$obj")
+        network=$(json network <<< "$obj")
+        prefix_l=$(json cidr <<< "$obj")
+
+        provision_start=$(get_provisionable_ip "${network}/${prefix_l}" start)
+        provision_end=$(get_provisionable_ip "${network}/${prefix_l}" end)
+
+        # shellcheck disable=SC2059
+        json=$(
+            printf "$json_tmpl" "$nic_tag" "$network_name" \
+                "$network" "$prefix_l" \
+                "$gateway" \
+                "$provision_start" \
+                "$provision_end" \
+                "$s_hostname" | json -o json-0
+        )
+
+        printf 'sdc-napi /networks -X POST -d '
+        printf \'
+        printf '%s' "$json"
+        printf \''\n'
+    done <<< "$ip_addresses"
+}
+
+function wait_for_state
+{
+    desired_state="$1"
+    current_state='unknown'
+
+    while [[ $current_state != "$desired_state" ]]; do
+        # Show some progress while waiting.
+        sleep 5 ; printf '.'
+        current_state=$(packet server get -j -i "$server" | json state)
+    done
+
+}
+
+##
+## Main
+##
+
+while getopts "h" options; do
+    case $options in
+        h) usage 0 ;;
+        *) echo "$OPTARG" ;;
+    esac
+done
+
+action="$1"
+
+if [[ -z $action ]]; then
+    usage 1
+fi
+
+shift
+
+case "$action" in
+    project)
+        do_setup_project "$@" ;;
+    headnode)
+        do_setup_server -n headnode "$@" ;;
+    cn|compute|computenode)
+        n="cn$(openssl rand -hex 3)"
+        do_setup_server -n "$n" "$@" ;;
+    sn)
+        hostname="$1"
+        uuid="$2"
+        show_napi_commands "${hostname:?}" "${uuid:?}"
+        ;;
+    *)
+        echo 'wut?' ;;
```

Comment:
A little less informal, perhaps?  Honestly I'd bail to usage here.

Reply:
Ha, forgot that was there. Fixed.